File: concrete/src/test/java/pcb/uwu/amount/quantity/AmperesTest.java
package pcb.uwu.amount.quantity;

import org.junit.Test;
import pcb.uwu.amount.composite.electromagnetism.Coulombs;
import pcb.uwu.amount.composite.electromagnetism.Ohms;
import pcb.uwu.amount.composite.electromagnetism.Siemens;
import pcb.uwu.amount.composite.electromagnetism.Volts;

import static java.math.MathContext.DECIMAL64;
import static org.junit.Assert.assertEquals;
import static pcb.uwu.core.Magnitude.KILO;
import static pcb.uwu.core.Magnitude.MILLI;

public class AmperesTest {
	private static final Amperes amperes = new Amperes(2, MILLI);

	@Test
	public void multipliedByTime() {
		assertEquals(new Coulombs(240, MILLI),
				amperes.multipliedBy(new Minutes(2), DECIMAL64));
	}

	@Test
	public void multipliedByElectricResistance() {
		assertEquals(new Volts(4),
				amperes.multipliedBy(new Ohms(2, KILO), DECIMAL64));
	}

	@Test
	public void dividedByElectricalPotential() {
		assertEquals(new Siemens(2),
				amperes.dividedBy(new Volts(1, MILLI), DECIMAL64));
	}

	@Test
	public void dividedByElectricalConductance() {
		assertEquals(new Volts(1, MILLI),
				amperes.dividedBy(new Siemens(2), DECIMAL64));
	}
}

File: implementation/src/test/java/pcb/uwu/amount/base/AmperesTest.kt
package pcb.uwu.amount.base

import pcb.uwu.amount.derived.electromagnetism.Coulombs
import pcb.uwu.amount.derived.electromagnetism.Ohms
import pcb.uwu.amount.derived.electromagnetism.Siemens
import pcb.uwu.amount.derived.electromagnetism.Volts
import pcb.uwu.core.Magnitude.KILO
import pcb.uwu.core.Magnitude.MILLI
import kotlin.test.Test
import kotlin.test.assertEquals

class AmperesTest
{
    @Test
    fun multipliedByTime() =
        assertEquals(expected = Coulombs(0.24),
                     actual = Amperes(2, MILLI) * Minutes(2))

    @Test
    fun multipliedByElectricResistance() =
        assertEquals(expected = Volts(4),
                     actual = Amperes(2, MILLI) * Ohms(2, KILO))

    @Test
    fun dividedByElectricalPotential() =
        assertEquals(expected = Siemens(2),
                     actual = Amperes(2, MILLI) / Volts(1, MILLI))

    @Test
    fun dividedByElectricalConductance() =
        assertEquals(expected = Volts("0.001"),
                     actual = Amperes(2, MILLI) / Siemens(2))
}